scala/scala.bzl

```diff
@@ -487,7 +498,10 @@ def _collect_jars_when_dependency_analyzer_is_off(dep_targets):
         compile_jars += dep_target.files
         runtime_jars += dep_target.files
 
-  return struct(compile_jars = compile_jars, transitive_runtime_jars = runtime_jars, jars2labels = {}, transitive_compile_jars = depset())
+  return struct(compile_jars = filter_not_sources(compile_jars),
```

Comment:
Why not remove them from runtime jars as well? Essentially I think this filter should be where we do "dep.files"

Reply:
Or rather only in the branch of "http file" this encapsulates the "mess" better and when we'll introduce scala import we can remove it

Reply:
I don't know if someone would want to do runtime reflection and access them tbh was the reason not to do it. Its mostly been accidental, and the likely hood of using/needing them at compile time seems amazingly low. But I don't mind dropping them from runtime, this just seemed smaller possible wake of unexpected behavior.

I don't think http_file branch would do it for us alone since we refer to maven_jar, which generates a build file. I'm not sure how easily we'll be able to remove it without saying that scala rules are incompatible with the native maven jar? (though this will only hit on some libraries where the sources path layout causes this path to fire too of course which is probably relatively uncommon -- this particular case cats are patching to remove).